crates/cxx-qt-gen/src/generator/rust/qobject.rs

```diff
@@ -116,6 +124,13 @@ fn generate_qobject_definitions(qobject_idents: &QObjectNames) -> Result<Generat
         }
     };
 
+    let base_upcast = if let Some(base) = base {
+        let base_name = type_names.lookup(&base)?.rust_unqualified();
+        quote! { impl cxx_qt::Upcast<#base_name> for #rust_struct_name_rust {} }
```

Comment:
this should be implemented on the Rust version of the C++ class not the Rust struct? And it needs to be qualified?

Eg this should be `impl cxx_qt::Upcast<qobject::Base> for qobject::MyObject {}` and not `impl cxx_qt::Upcast<Base> for MyObjectRust {}`